pkg/api/endpoints/util.go

```diff
@@ -33,31 +34,35 @@ import (
 // form for things like comparison.  The result is a newly allocated slice.
 func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 	// First map each unique port definition to the sets of hosts that
-	// offer it.  The sets of hosts must be de-duped, using IP as the key.
-	type ipString string
-	allAddrs := map[ipString]*api.EndpointAddress{}
+	// offer it.  The sets of hosts must be de-duped, using IP+UID as the key.
+	type addressKey struct {
```

Comment:
This needs unit tests to demonstrate the expected behavior and ensure it stays working.